blob/azureblob/azureblob.go

```diff
@@ -161,7 +161,7 @@ func (o *lazyCredsOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.
 
 		isMSIEnvironment := adal.MSIAvailable(ctx, adal.CreateSender())
 
-		if accountKey != "" {
+		if accountKey != "" || sasToken != "" {
```

Comment:
I didn't see existing tests for this `OpenBucketURL`. Not sure if it's easy to do without mocking these `opener` calls.

Reply:
There are tests for the URL parsing etc. (https://github.com/google/go-cloud/blob/master/blob/azureblob/azureblob_test.go#L527) but we don't have good tests for the various authentication mechanisms unfortunately.